ARCHITECTURE.md

```diff
@@ -0,0 +1,191 @@
+cddl-gen Architecture
+=====================
+
+Since cddl-gen is a Python script that generates C code, this document is split into two sections:
+
+1. Architecture of the Python script
+2. Architecture of the generated code
+
+Architecture of the Python script
+=================================
+
+The cddl_gen.py script is located in [scripts/cddl_gen.py](scripts/cddl_gen.py).
+
+The functionality is spread across 5 classes:
+
+1. CddlParser
+2. CddlXcoder (inherits from CddlParser)
+3. PythonClass (inherits from CddlXcoder)
+4. CodeGenerator (inherits from CddlXcoder)
+5. CodeRenderer
+
+CddlParser
+----------
+
+Each CddlParser object represents a CDDL type.
+Since CDDL types can contain other types, CddlParser recursively parses a CDDL string, and spawns new instances of itself to represent the (child) types it contains.
```

Comment:
"Child" makes the hierarchical relationship more clear IMO. I'll think some more about this.

Reply:
I will stick with "child" for now, as it would be hard to change it everywhere, and I don't exactly know what I would replace it with in all cases. I added a sentence mentioning what "children" refers to.